pkg/daemon/update_test.go

```diff
@@ -318,16 +333,81 @@ func TestUpdateSSHKeys(t *testing.T) {
 	}
 }
 
+// This test should fail until Ignition validation enabled.
+// Ignition validation does not permit writing files to relative paths.
+func TestInvalidIgnConfig(t *testing.T) {
+	// expectedError is the error we will use when expecting an error to return
+	expectedError := fmt.Errorf("broken")
+	// testClient is the NodeUpdaterClient mock instance that will front
+	// calls to update the host.
+	testClient := RpmOstreeClientMock{
+		GetBootedOSImageURLReturns: []GetBootedOSImageURLReturn{},
+		RunPivotReturns: []error{
+			// First run will return no error
+			nil,
+			// Second rrun will return our expected error
+			expectedError},
+	}
+	mockFS := &FsClientMock{MkdirAllReturns: []error{nil}, WriteFileReturns: []error{nil}}
+	// Create a Daemon instance with mocked clients
+	d := Daemon{
+		name:              "nodeName",
+		OperatingSystem:   machineConfigDaemonOSRHCOS,
+		NodeUpdaterClient: testClient,
+		loginClient:       nil, // set to nil as it will not be used within tests
+		client:            fake.NewSimpleClientset(),
+		kubeClient:        k8sfake.NewSimpleClientset(),
+		rootMount:         "/",
+		bootedOSImageURL:  "test",
+		fileSystemClient:  mockFS,
+	}
+
+	oldMcfg := &mcfgv1.MachineConfig{
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: ignv2_2types.Config{
+				Ignition: ignv2_2types.Ignition{
+					Version: "2.2.0",
+				},
+			},
+		},
+	}
+	// create file to write that contains an impermissable relative path
+	tempFileContents := ignv2_2types.FileContents{Source: "data:,hello%20world%0A"}
+	tempMode := 420
+	newMcfg := &mcfgv1.MachineConfig{
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: ignv2_2types.Config{
+				Ignition: ignv2_2types.Ignition{
```

Comment:
As a follow up, I would create another failing case where the MC doesn't have the ignition version and it correctly fails. Not for this PR.

Reply:
sure thing!